packages/amazon-bedrock/src/bedrock-provider.ts

```diff
@@ -63,42 +88,33 @@ Create an Amazon Bedrock provider instance.
 export function createAmazonBedrock(
   options: AmazonBedrockProviderSettings = {},
 ): AmazonBedrockProvider {
-  const createBedrockRuntimeClient = () =>
-    new BedrockRuntimeClient(
-      options.bedrockOptions ?? {
-        region: loadSetting({
+  const sign = createSigV4SigningFunction({
+    region: options.region,
+    accessKeyId: options.accessKeyId,
+    secretAccessKey: options.secretAccessKey,
+    sessionToken: options.sessionToken,
+  });
+
+  const getBaseUrl = (): string =>
+    withoutTrailingSlash(
+      options.baseURL ??
+        `https://bedrock-runtime.${loadSetting({
           settingValue: options.region,
           settingName: 'region',
           environmentVariableName: 'AWS_REGION',
           description: 'AWS region',
-        }),
-        credentials: {
-          accessKeyId: loadSetting({
-            settingValue: options.accessKeyId,
-            settingName: 'accessKeyId',
-            environmentVariableName: 'AWS_ACCESS_KEY_ID',
-            description: 'AWS access key ID',
-          }),
-          secretAccessKey: loadSetting({
-            settingValue: options.secretAccessKey,
-            settingName: 'secretAccessKey',
-            environmentVariableName: 'AWS_SECRET_ACCESS_KEY',
-            description: 'AWS secret access key',
-          }),
-          sessionToken: loadOptionalSetting({
-            settingValue: options.sessionToken,
-            environmentVariableName: 'AWS_SESSION_TOKEN',
-          }),
-        },
-      },
```

Comment:
I still feel we are delaying errors by pushing this down but could be wrong.

Reply:
Let me look into moving it to happen earlier, we could attempt to build credentials pretty much straightaway, executing these settings lookups, and then pass them to where they're needed.

There are two things that I think it will hit:
- default provider instance use in unit tests where we don't have the env vars defined, but as we've discussed we could shift those to instantiate the model directly (I already did, this may have been resolved).
- example scripts where we may import the provider (and thus instantiate the default instance) before we import `dotenv/config`. Not sure what to do about this one.

Reply:
> * example scripts where we may import the provider (and thus instantiate the default instance) before we import `dotenv/config`. Not sure what to do about this one.

We could edit the default-provider pattern we use to lazily construct the instance. Arguably executing settings-load-lookup on import of code before any instance is referenced or method/fn is called is a little premature. I will try this and see how it looks.